frontend/components/search.js

```diff
@@ -68,22 +78,27 @@ class Search {
 
     if (results.error) {
       macros.error('Error with networking request', results.error);
-      return [];
+      return { results: [] };
     }
 
-
+    // if cache doesn't exist, instantiate. Subject info only changed here
+    // since it should only be changed on cache misses
     if (!this.cache[termId + query]) {
-      this.cache[termId + query] = [];
+      this.cache[termId + query] = {};
+      this.cache[termId + query].results = [];
+      this.cache[termId + query].subjectName = waitedRequest.subjectName;
+      this.cache[termId + query].subjectCount = waitedRequest.subjectCount;
     }
 
     // Add to the end of exiting results.
-    this.cache[termId + query] = this.cache[termId + query].concat(results);
+    this.cache[termId + query].results = this.cache[termId + query].results.concat(results);
+
 
     if (results.length < termCount - existingTermCount) {
       this.allLoaded[termId + query] = true;
     }
 
-    return this.cache[termId + query];
+    return (this.cache[termId + query]);
```

Comment:
```suggestion
    return this.cache[termId + query];
```
Looks like you're returning a tuple otherwise.